Backpack-SwiftUI/CardCarousel/Classes/BPKCardCarousel.swift

```diff
@@ -0,0 +1,173 @@
+import SwiftUI
+
+struct BPKCardCarousel<Content: View>: View {
+    let cards: [Content]
+    let startingIndex: Int
+    let onCardChange: () -> Void
+    
+    init(
+        cards: [Content],
+        startingIndex: Int = 1,
+        onCardChange: @escaping () -> Void = { }
+    ) {
+        self.cards = cards
+        self.startingIndex = startingIndex
+        self.onCardChange = onCardChange
+    }
+    
+    var body: some View {
+        GeometryReader { reader in
+            InternalCardCarousel(
+                size: reader.size,
+                content: cards,
+                startingIndex: startingIndex,
+                onCardChange: onCardChange
+            )
+                .frame(
+                    width: reader.size.width,
+                    height: reader.size.height)
+        }
+    }
+}
+
+internal struct InternalCardCarousel<Content: View>: View {
+    private let size: CGSize
+    private let cardCount: Int
+    private let cards: [Content]
+    private let onCardChange: () -> Void
+    
+    private let dragAnimation: Animation = .snappy(duration: 0.5)
+    private let cardWidth: CGFloat
+    
+    @State private var currentCardIndex: Int
```

Comment:
This shall be a Binding instead and have this value exposed to the parent, so that the client then has access to this property and it's the client that's in control over what index is selected.